src/vs/workbench/browser/parts/titlebar/menubarControl.ts

```diff
@@ -617,8 +623,9 @@ export class CustomMenubarControl extends MenubarControl {
 						}
 
 						const submenuActions: SubmenuAction[] = [];
-						updateActions(submenu, submenuActions, topLevelTitle);
-						target.push(new SubmenuAction(action.id, mnemonicMenuLabel(title), submenuActions));
+						if (updateActions(submenu, submenuActions, topLevelTitle)) {
```

Comment:
same as comment below, we know the submenuActions length after running updateActions, why tie it to a boolean that makes it unclear what we are checking for?

Reply:
To be honest: because the code was hard to understand and it was the only way I could figure out how to do it. Will push a commit with the suggested changes.